datalad/support/gitrepo.py

```diff
@@ -3444,6 +3466,63 @@ def save_(self, message=None, paths=None, _status=None, **kwargs):
         # TODO yield result for commit, prev helper checked hexsha pre
         # and post...
 
+    def _check_for_win_compat(self, files, config):
+        """Check file names for illegal characters or reserved names on Windows
+
+        In the case that a non-Windows-compatible file is detected, warn users
+        about potential interoperability issues.
+
+        Parameters
+        ----------
+        files
+          list of files to add
+        config
+          value of self.config.obtain("datalad.save.windows-compat-warning"),
+          used to choose appropriate behavior. "none" performs no check,
+          "warning" warns in case of incompatibilities, and "error" results in
+          an error result in case of incompatibilities
+        """
+        # don't perform any check when the configuration is set to 'none'
+        if config == 'none':
+            return files, None
+
+        from collections import defaultdict
+        problems = defaultdict(list)
+        for file in files:
+            for part in Path(file).parts:
```

Comment:
another concern/question came to my mind -- since you are going through parts, what if we have a folder which has forbidden name -- this will report for every file under that folder instead of the folder itself right? (e.g. folder `con/` having good number of files under it).  I think only the folder should be reported (once), so below not `file` should be added to `paths` but rather the path composed from the parts considered, and only if was not seen before, right?

Reply:
From the top of my head I *think* this is handled below in 

```
for k, v in problems.items():

+ # use the key as an explanation, and report filenames only once

+ msg += f"\n{k} {[*{*v}]}" if k != 'paths' else ''
```

But I will check 👍

Reply:
Its all behaving nicely:

```
(handbook2) adina@muninn in /tmp/123 on git:master
❱ tree
.
└── con
    ├── file1
    ├── file2
    ├── file3
    └── illegal?

1 directory, 4 files
(handbook2) adina@muninn in /tmp/123 on git:master
❱ datalad save .
[WARNING] Some elements of your dataset are not compatible with Windows systems. Disable this check by changing datalad.save.windows-compat-warning or consider renaming the following elements: 
| Elements using a reserved filename: ['con']
| Elements with illegal characters: ['illegal?'] 
add(ok): con/file1 (file)                                                       
add(ok): con/file2 (file)                                                       
add(ok): con/file3 (file)                                                       
add(ok): con/illegal? (file)                                                    
save(ok): . (dataset)      
```

Reply:
oh sweet! well done!